extensions/ipynb/src/notebookImagePaste.ts

```diff
@@ -76,9 +84,9 @@ class CopyPasteEditProvider implements vscode.DocumentPasteEditProvider {
 				}
 			}
 			currentCell.metadata.custom.attachments[pasteFilename] = { 'image/png': b64string };
+			metadataNotebookEdit = vscode.NotebookEdit.updateCellMetadata(currentCell.index, currentCell.metadata);
 		}
 
-		const metadataNotebookEdit = vscode.NotebookEdit.updateCellMetadata(currentCell.index, currentCell.metadata);
```

Comment:
Would it be clearer to extract this to new function that computes the new metadata? That way you only need a single call to `updateCellMetadata`

Reply:
Ah maybe you'd also need to return the `pasteFilename` in this case. Having the duplicate calls in a long function though suggests this should be split up

Reply:
refactored into own cell, with return parameter being the correct type for metadata. Not sure what you meant by returning `pasteFilename`, I might have just gotten around that with how the method constructs and returns the finalized metadata update.